### PR TITLE
feat(profiling): minimap resize

### PR DESCRIPTION
**Pull request by eliashussary**
## Summary
This PR adds the ability to resize the miniMap position.

Closes: https://github.com/getsentry/team-profiling/issues/60

https://user-images.githubusercontent.com/7349258/204390968-f2d8cc80-a978-4cea-9241-ec9a3c119fc5.mov

### CODE REVIEW

File: static/app/utils/profiling/gl/utils.ts [hunk 1]
```diff
@@ -682,3 +682,85 @@ export function computeConfigViewWithStategy(
 
   return frame.withHeight(view.height);
 }
+
+export class RectCursorUtil {
```
Comment (eliashussary) [inline static/app/utils/profiling/gl/utils.ts:686]: i made this a standalone class but don't have strong opinions about it, these could all be standalone util functions. Some of the methods aren't actually used, but i filled them out for completeness.

The gist of what this class does is creates two 'buffered' virtual rects in and around the actual rect that are (in this case the mini map view).

The area between the outer rect and the inner rect becomes our resize window area... rect's our border.

The inner area becomes our grab area to drag.


image

Comment (JonasBa) [reply static/app/utils/profiling/gl/utils.ts:686]: I think we can avoid creating a new rect for each mouse move and just compute the distance from the edges to the cursor. One way of doing this without doing a bounds check would be to take the min value of the absolute distance between the edges and the cursor. Something like min(abs(left edge - cursor x), abs(right edge - cursor x)) would give you either the left or right edge that you are closest to. Similar with min(abs(top edge - cursor y), abs(bottom edge - cursor y)) would give you top/bottom edge. If you take the min distance of those 4, you get the edge that you are nearest to. 

The benefit of that is that you are being more precise as you always take a min distance (which could be the case when the rect is very small).

File: static/app/components/profiling/flamegraphZoomViewMinimap.tsx [hunk 2]
```diff
@@ -227,6 +237,7 @@ function FlamegraphZoomViewMinimap({
     [flamegraphMiniMapCanvas, flamegraphMiniMapView, lastDragVector, canvasPoolManager]
   );
 
+  const prevConfigSpaceCursor = useRef(vec2.fromValues(0, 0));
```
Comment (JonasBa) [inline static/app/components/profiling/flamegraphZoomViewMinimap.tsx:240]: Damn, if mousemove had deltaX or something it would be ❤️

Comment (JonasBa) [reply static/app/components/profiling/flamegraphZoomViewMinimap.tsx:240]: On a second note, should this be vec2|null type?

File: static/app/components/profiling/flamegraphZoomViewMinimap.tsx [hunk 3]
```diff
@@ -424,46 +474,49 @@ function FlamegraphZoomViewMinimap({
...
       />
       <OverlayCanvas ref={c => setFlamegraphMiniMapOverlayCanvasRef(c)} />
     </Fragment>
   );
 }
 
+function getCanvasCursor(
+  configView: Rect | undefined,
+  configSpaceCursor: vec2 | null,
+  borderSize: number
```
Comment (JonasBa) [inline static/app/components/profiling/flamegraphZoomViewMinimap.tsx:498]: borderWidth?

File: static/app/components/profiling/flamegraphZoomViewMinimap.tsx [hunk 4]
```diff
@@ -245,6 +256,27 @@ function FlamegraphZoomViewMinimap({
         setLastInteraction('pan');
         return;
       }
+
+      if (evt.buttons === 1 && lastInteraction === 'resize') {
```
Comment (JonasBa) [inline static/app/components/profiling/flamegraphZoomViewMinimap.tsx:260]: Why do we check for evt.buttons here? Are we trying to check that the user is holding the button?

**Review by JonasBa — APPROVED** [review]
Nothing blocking, nice work

File: static/app/components/profiling/flamegraphZoomViewMinimap.tsx (hunk 4 above)
Comment (eliashussary) [reply static/app/components/profiling/flamegraphZoomViewMinimap.tsx:260]: Yeah its to determine if mousedown is still being held.  I don't think we need it anymore and will drop it. Pretty sure i threw it in there when i was dealing w/ that issue where i was losing the mouse interaction... at that time i was piggy backing off of the `select` interaction and shot myself in the foot.

**Pull request merged**